domain/common.go

```diff
@@ -70,3 +70,9 @@ type Prereq struct {
 	Name   string
 	Script string
 }
+
+const (
```

Comment:
We shouldn't duplicate these strings (here and in each of the `kind` variables in the various entity-specific subdirectories).  

I suggest adding a GetType() method to each domain package in addition to the GetType() for each domain object. For instance, something like this for ResourcePool:
```
func (p *ResourcePool) GetType() string {
	return GetType()
}


func GetType() string {
	return kind
}
```